src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,468 @@
+      itemToCreate.put("permanentLocationId", permLocation.toString());
+    }
+
+    return itemToCreate.mapTo(Item.class);
+  }
+
+  private Item getItem(String id) throws Exception {
+    return itemsClient.getById(UUID.fromString(id)).getJson().mapTo(Item.class);
+  }
+
+  private Items findItems(String searchQuery) throws Exception {
+    CompletableFuture<Response> searchCompleted = new CompletableFuture<>();
+
+    client.get(itemsStorageUrl("?query=") + urlEncode(searchQuery),
+      StorageTestSuite.TENANT_ID, ResponseHandler.json(searchCompleted));
+
+    return searchCompleted.get(5, TimeUnit.SECONDS).getJson()
+      .mapTo(Items.class);
+  }
+
+  private IndividualResource createItem(Item item) throws Exception {
+    return itemsClient.create(JsonObject.mapFrom(item));
+  }
+
+  // is actually used by @Parameters(method = "parameters")
+  @SuppressWarnings("unused")
+  private List<String[]> parameters() {
+    List<String[]> holdingLocations = Arrays.asList(
+      new String[]{null, null},
+      new String[]{null, MAIN_LIBRARY_LOCATION},
+      new String[]{SECOND_FLOOR_LOCATION, null},
+      new String[]{SECOND_FLOOR_LOCATION, MAIN_LIBRARY_LOCATION}
+    );
+
+    List<String[]> itemLocations = Arrays.asList(
+      new String[]{null, null},
+      new String[]{null, THIRD_FLOOR_LOCATION},
+      new String[]{FOURTH_FLOOR_LOCATION, null},
+      new String[]{FOURTH_FLOOR_LOCATION, THIRD_FLOOR_LOCATION}
+    );
+
+    List<String[]> newItemLocations = Arrays.asList(
+      new String[]{null, null},
+      new String[]{null, THIRD_FLOOR_LOCATION},
+      new String[]{THIRD_FLOOR_LOCATION, null},
+      new String[]{THIRD_FLOOR_LOCATION, FOURTH_FLOOR_LOCATION},
+      new String[]{null, ANNEX_LIBRARY_LOCATION},
+      new String[]{ANNEX_LIBRARY_LOCATION, null},
+      new String[]{ANNEX_LIBRARY_LOCATION, ONLINE_LOCATION}
+    );
+
+    // Combine the locations
+    List<String[]> parameters = new ArrayList<>();
+    for (String[] holdingsLocation : holdingLocations) {
+      for (String[] itemLocation : itemLocations) {
+        for (String[] newItemLocation : newItemLocations) {
+
+          parameters.add(new String[]{
+            holdingsLocation[0], holdingsLocation[1],
+            itemLocation[0], itemLocation[1],
+            newItemLocation[1], newItemLocation[1]
+          });
+        }
+      }
+    }
+
+    return parameters;
+  }
+
+  private UUID lookupLocationByName(String name)
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+    if (name == null) {
+      return null;
+    }
+
+    return locationsClient.getAll().stream()
+      .filter(json -> json.getString("name").equals(name))
+      .map(json -> UUID.fromString(json.getString("id")))
+      .findAny()
+      .orElse(null);
+  }
+
+  private void setPermanentTemporaryLocation(
+    JsonObject json, UUID permanentLocationId, UUID tempLocationId) {
+    String permanentLocationIdFieldName = "permanentLocationId";
```

Comment:
Are these constants? If so, they should be marked final.

Personally, I might find it easier to read if there was whitespace between the signature and this first line.

Reply:
Fixed.